### DISCUSSION TITLE
Dialog buttons when updating a dialog with new content

### DISCUSSION THREAD

**Discussion by peavine** [swiftDialog/swiftDialog#471] [General]
I'm working to learn a little about swiftDialog's ability to update a dialog with new content. For testing purposes (not for actual use), I wrote the following AppleScript, which generally works as I want. However, I want to add a "Sleep" button in case the user does not want to wait 10 seconds. Is there some way to do this within the context of updating a dialog with new content? I read the Wiki on this topic but didn't see anything relevant. Thanks for any pointers. 

```
set dialogPID to do shell script "/usr/local/bin/dialog --title none --message 'The computer will sleep in 10 seconds' --messagefont 'size=14' --width 320 --height 75 --button1text 'Cancel' --hideicon &> /dev/null & echo $!"

repeat with i from 9 to 0 by -1
	delay 1
	if dialogExists(dialogPID) is false then error number -128
	set theMessage to "The computer will sleep in " & i & " seconds."
	do shell script "/bin/echo 'message: " & theMessage & "' >> /var/tmp/dialog.log"
end repeat

do shell script "/bin/echo 'quit: true' >> /var/tmp/dialog.log"

--sleep command goes here

on dialogExists(dialogPID)
	try
		do shell script "ps -p " & dialogPID & " > /dev/null"
		return true
	end try
	return false
end dialogExists
```

**Comment by bartreardon** [comment]
just have a button that you check the exit code for. if you see the exit code for the associated button (0 or 2 or 3) then cancel your sleep timer and go straight into whatever process you want to do.

**Comment by peavine** [comment]
Thanks Bart. I'll work on that.
